src/poly/poly.jl

```diff
@@ -1007,22 +1007,23 @@ function evaluate(a::SPolyUnion{T}, C::Vector{n_Z}) where T <: Nemo.RingElem
    return evaluate(a, C2)
 end
 
-function (a::SPolyUnion{T})(vals::T...) where T <: RingElem
-   length(vals) != nvars(parent(a)) && error("Number of variables does not match number o
-f values")
-   return evaluate(a, [vals...])
+function (a::SPolyUnion{T})() where T <: RingElem
+   0 != nvars(parent(a)) && error("Number of variables does not match number of values")
+   return evaluate(a, T[])
 end
 
-function (a::SPolyUnion{T})(vals::U...) where {T <: RingElem, U <: Union{Integer, Rational,
```

Comment:
Since this PR does not seem to resolve all ambiguities (judging by the TODO comment below) perhaps it would be best to not touch this one for the time being -- I am not keen of this kind of code duplication and would prefer to solve it differently.  E.g., perhaps the ambiguity can also be resolved by removing the first method and changing this one to use `U <: Union{T, Integer, ...` ?

But I am not expecting you to figure that out, hence my suggestion to perhaps just drop this for now

Reply:
This does indeed fix all errors of type `unbound_args` by Aqua. In this case here, when you input a zero-length vararg, U ist unbound.

But if so wished for, I can move 93e15935302f839fb93c045f6a4a843a445c69dd into its own PR later today.